### PR TITLE
Check oracle response Content-Type header

### LINKED ISSUE #2036: Add support for OracleResponseCode:ContentTypeNotSupported
See neo-project/neo#2505.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #2036 .

### CODE REVIEW

File: pkg/services/oracle/request.go [hunk 1]
```diff
@@ -126,6 +127,11 @@ func (o *Oracle) processRequest(priv *keys.PrivateKey, req request) error {
 			}
 			switch r.StatusCode {
 			case http.StatusOK:
+				if !checkMediaType(r.Header.Get("Content-Type"), o.MainCfg.AllowedContentTypes) {
+					resp.Code = transaction.ContentTypeNotSupported
```
Comment (AnnaShaleva) [inline pkg/services/oracle/request.go:132]: Don't we want to log a warning? It's done for ProtocolNotSupported.

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/services/oracle/request.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/services/oracle/request.go:132]: `ProtocolNotSupported` is an invalid request, `ContentTypeNotSupported` is an invalid reply. We usually don't log these (it's not much different from Forbidden/NotFound/...).

File: pkg/config/oracle_config.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import "time"
 type OracleConfiguration struct {
 	Enabled               bool               `yaml:"Enabled"`
 	AllowPrivateHost      bool               `yaml:"AllowPrivateHost"`
+	AllowedContentTypes   []string           `yaml:"AllowedContentTypes"`
```
Comment (roman-khimov) [inline pkg/config/oracle_config.go:9]: We need our default configurations to contain this as well as `docs/oracle.md`.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: pkg/config/oracle_config.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/config/oracle_config.go:9]: But what about testnet/mainnet? These are technically limited too and it's even more important than for private networks.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**